packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -363,6 +361,17 @@ class _CupertinoSheetTransitionState extends State<CupertinoSheetTransition> {
     }
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    SystemChrome.setSystemUIOverlayStyle(
```

Comment:
Does this need to be called every time the widget rebuilds?

> should change SystemUIOverlayStyle based on theme

How is theme accounted for?

Reply:
I wonder because it does not look like it changes or depends on anything. Would `initState` one and done be better?

Reply:
> How is theme accounted for?
It was in this PR originally, but that wasn't the correct behavior. It should set a static system overlay style, and leave it up to the previous page to overwrite it. See: https://github.com/flutter/flutter/pull/164680#pullrequestreview-2734079578

Reply:
> I wonder because it does not look like it changes or depends on anything. Would initState one and done be better?

Yaa, even i was thinking about it. Because now we don't use theme data for this change.